src/Xamarin.Android.Build.Tasks/Tasks/ResolveLibraryProjectImports.cs

```diff
@@ -302,11 +314,11 @@ static string GetTargetAssembly (ITaskItem assemblyName)
 										.Replace ("library_project_imports\\","")
 										.Replace ("library_project_imports/", "");
 									if (path.EndsWith (".jar", StringComparison.OrdinalIgnoreCase)) {
-										AddJar (jars, importsDir, path);
+										AddJar (jars, importsDir, path, assemblyPath);
 									}
 									return path;
 								}, deleteCallback: (fileToDelete) => {
-									return !jars.Contains (fileToDelete);
+									return !jars.Any (x => x.ItemSpec == fileToDelete);
```

Comment:
If we are having to do `.Any ()`, should we make the list of jars a `Dictionary<string, ITaskItem>`?